pennylane/devices/preprocess.py

```diff
@@ -253,6 +255,7 @@ def decompose(
     ] = None,
     max_expansion: Union[int, None] = None,
     name: str = "device",
+    error: Exception = DeviceError,
```

Comment:
Mind adding this to the list of Args too? Potentially we should break all the optional kwargs into a `Keyword Args:` section at this point.

Reply:
Can't comment on unchanged code, but could you also change the hard-coded ``DecompositionUndefinedError`` mentions in the `Args:` section?